shared_model/interfaces/transaction_responses/unknown_tx_response.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_UNKNOWN_TX_RESPONSE_HPP
+#define IROHA_UNKNOWN_TX_RESPONSE_HPP
+
+#include "interfaces/primitive.hpp"
+
+namespace shared_model {
+  namespace interface {
+    /**
+     * Transaction not found
+     */
+    class UnknownTxResponse : public AbstractTxResponse<UnknownTxResponse> {
+     private:
+      std::string className() const override { return "UnknownTxResponse"; }
+
+      iroha::model::TransactionResponse::Status oldModelStatus() const {
+        return iroha::model::TransactionResponse::Status::
+            NOT_RECEIVED;
```

Comment:
Is it a good mapping `unknown <-> not_received`? May you explain that a bit

Reply:
The idea is not_received possible to achieve in two situations: when transaction with those hash not received and transaction received but was rejected long time ago. `NOT_RECEIVED` not reflected name for the second case. For my opinion unknow covered both, but you may provide a more suitable name.